gateway/src/apicast/policy/3scale_batcher/3scale_batcher.lua

```diff
@@ -111,16 +118,37 @@ local function error(service, rejection_reason)
 end
 
 local function handle_backend_ok(self, transaction)
+  if self.cache_handler then
+    self.backend_downtime_cache:update(transaction, 200, self.cache_handler)
+  end
+
   self.auths_cache:set(transaction, 200)
   self.reports_batcher:add(transaction)
 end
 
 local function handle_backend_denied(self, service, transaction, status, headers)
+  if self.cache_handler then
+    self.backend_downtime_cache:update(transaction, status, self.cache_handler)
+  end
+
   local rejection_reason = rejection_reason_from_headers(headers)
   self.auths_cache:set(transaction, status, rejection_reason)
   return error(service, rejection_reason)
 end
 
+local function handle_backend_error(self, service, transaction)
+  local cached = self.cache_handler and
+      self.backend_downtime_cache:get(transaction)
+
+  if not cached or cached ~= 200 then
```

Comment:
Might be better to do first the positive check ? `if cached == 200 then ... else ... end`

Reply:
Changed. Looks better :+1: 

I was thinking in the case of `cached` being a table, that's why I added that nil check, but it's an integer, so there's no need to do that.